pkg/telemetry/context.go

```diff
@@ -0,0 +1,50 @@
+package telemetry
+
+import (
+	"context"
+	"strings"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"google.golang.org/grpc"
+)
+
+type rpcContextName string
+
+const (
+	methodContextName  rpcContextName = "method"
+	serviceContextName rpcContextName = "service"
+)
+
+func saveMethodServiceInContext(ctx context.Context, method string, service string) context.Context {
+	methods := strings.Split(method, "/")
+
+	ctx = context.WithValue(ctx, methodContextName, methods[len(methods)-1])
```

Comment:
Instead of enumerating per field just stuff a struct in here. That makes it more extensible in the future as well.

```go
type RPCInfo struct {
    Method string
    Service string
}

func WithRPCInfo(ctx context.Context, info RPCInfo) {
    return context.WithValue(ctx, rpcInfoContextKey, info)
}

func RPCInfoFromContext(ctx context.Context) RPCInfo {
    rpcInfo, ok := ctx.Value(rpcInfoContextKey).(RPCInfo)
    ...

    return rpcInfo
}
```